bugsnag-plugin-android-ndk/src/main/java/com/bugsnag/android/ndk/OpaqueValue.kt

```diff
@@ -0,0 +1,45 @@
+package com.bugsnag.android.ndk
+
+import com.bugsnag.android.JsonStream
+import java.io.StringWriter
+
+/**
+ * Marker class for values that are `BSG_METADATA_OPAQUE_VALUE` in the C layer
+ */
+internal class OpaqueValue(val json: String) {
+    companion object {
+        private const val MAX_NDK_STRING_LENGTH = 64
+        private const val US_ASCII_MAX_CODEPOINT = 128
```

Comment:
ASCII max codepoint is actually 127, but you also wouldn't want codepoint 127 in a string because it's the DEL control code.